pyomo/repn/quadratic.py

```diff
@@ -157,11 +158,15 @@ def append(self, other):
                 self.nonlinear += nl
 
 
-def _mul_linear_linear(varOrder, linear1, linear2):
+def _mul_linear_linear(visitor, linear1, linear2):
     quadratic = {}
+    vo = visitor.var_recorder.var_order
     for vid1, coef1 in linear1.items():
         for vid2, coef2 in linear2.items():
-            if varOrder(vid1) < varOrder(vid2):
+            # Note that this is random.  If the client cares about
+            # determinism, it may need to reverse the keys based on
+            # something more deterministic than vid
```

Comment:
Do we not care about enforcing determinism here from the dev side?

Reply:
Same question as Miranda...

Reply:
The comment it wrong.  This is sorting based on the identified variable order, which should be deterministic (for deterministic input)